bigquery_etl/public_data/publish_gcs_metadata.py

```diff
@@ -0,0 +1,202 @@
+"""Generate and upload JSON metadata files for public datasets on GCS."""
+
+from argparse import ArgumentParser
+import json
+import logging
+import os
+import re
+import smart_open
+
+from google.cloud import storage
+from itertools import groupby
+
+from ..parse_metadata import Metadata
+from ..util import standard_args
+
+
+DEFAULT_BUCKET = "mozilla-public-data-http"
+DEFAULT_API_VERSION = "v1"
+DEFAULT_ENDPOINT = "http.public-data.prod.dataops.mozgcp.net/"
+REVIEW_LINK = "https://bugzilla.mozilla.org/show_bug.cgi?id="
+GCS_FILE_PATH_RE = re.compile(
+    r"api/(?P<api_version>.+)/tables/(?P<dataset>.+)/(?P<table>.+)/(?P<version>.+)/"
+    r"files/(?:(?P<date>.+)/)?(?P<filename>.+\.json\.gz)"
+)
+
+parser = ArgumentParser(description=__doc__)
+parser.add_argument(
+    "--project_id", "--project-id", default="mozilla-public-data", help="Target project"
+)
+parser.add_argument(
+    "--target-bucket",
+    "--target_bucket",
+    default=DEFAULT_BUCKET,
+    help="GCP bucket JSON data is exported to",
+)
+parser.add_argument(
+    "--endpoint", default=DEFAULT_ENDPOINT, help="The URL to access the HTTP endpoint"
+)
+parser.add_argument(
+    "--api_version",
+    "--api-version",
+    default=DEFAULT_API_VERSION,
+    help="Endpoint API version",
+)
+parser.add_argument(
+    "--target", help="File or directory containing metadata files", default="sql/"
+)
+standard_args.add_log_level(parser)
+
+
+class GcsTableMetadata:
+    """Metadata associated with table data stored on GCS."""
+
+    def __init__(self, files, endpoint, target_dir):
+        """Initialize container for metadata of a table published on GCS."""
+        assert len(files) > 0
+        self.files = files
+        self.endpoint = endpoint
+        self.files_path = self.files[0].split("files")[0] + "files"
+        self.files_uri = endpoint + self.files_path
+
+        (self.dataset, self.table, self.version) = dataset_table_version_from_gcs_path(
+            self.files[0]
+        )
+        self.metadata = Metadata.of_table(
+            self.dataset, self.table, self.version, target_dir
+        )
+
+    def table_metadata_to_json(self):
+        """Return a JSON object of the table metadata for GCS."""
+        metadata_json = {}
+        metadata_json["friendly_name"] = self.metadata.friendly_name
+        metadata_json["description"] = self.metadata.description
+        metadata_json["incremental"] = self.metadata.is_incremental()
+        metadata_json["incremental_export"] = self.metadata.is_incremental_export()
+
+        if self.metadata.review_bug() is not None:
+            metadata_json["review_link"] = REVIEW_LINK + self.metadata.review_bug()
+
+        metadata_json["files_uri"] = self.files_uri
+        # todo: add last updated
```

Comment:
I'll open an issue for publishing information about the last time table data was updated